lib/puppet/ssl/validator/default_validator.rb

```diff
@@ -118,9 +114,9 @@ def call(preverify_ok, store_context)
   # @api private
   #
   def setup_connection(connection)
-    if ssl_certificates_are_present?
+    if @ca_path
```

Comment:
Followup to the above, alternatively, do we want this to be a check for whether this file exists, rather than that the variable has been set? A quick grep of the code shows us always creating the default validator with no args, which means I think this will always evaluate to `true` unless someone deliberately unset the settings above.

Reply:
Does this check need to happen when we set up every connection? Or can it only be done once per validator?

Reply:
I moved this back to the previous `#ssl_certificates_are_present?` implementation.

This will be executed—eventually—every time we create a new connection [here](https://github.com/puppetlabs/puppet/blob/master/lib/puppet/network/http_pool.rb). As @Magisus pointed out in a separate comment, the code checks whether or not the files exist with `Puppet::FileSystem.exist?` to determine the `verify_mode` of the connection.

I'm unclear what the implications are for moving this code outside of this method with the respect to what happens during an agent run. Probably the auto-signing behavior is in play here?